### ISSUE TITLE
Change mode

### ISSUE THREAD

**Issue by kavillock** [peruihkxt/brackets-smarty#1]
Hi, i like this plug-in but is a chance to change mode smarty to smartymixed

**Comment by bendrexl** [comment]
Mixed mode would be nice - I'm trying to use this with OpenCart's templates, which are mostly PHP with just a little Smarty.

**Comment by kavillock** [comment]
yes, but is also an error with codemirror, https://github.com/codemirror/CodeMirror/issues/2871

**Comment by peruihkxt** [comment]
@kavillock Thanks for that link. I finally (hopefully) have some time to put toward this. I think the way to go is to specify a base mode for CodeMirror, but I don't think Brackets exposes that parameter in its API.

An additional problem is detecting the type of base (output) file, since it can be PHP, HTML, Javascript, and possibly others, although PHP is probably the most common use case.

I think it needs to change the CM language whenever a file is opened. If y'all want to submit a PR or suggest what APIs to use, I'll try to get this working better.
